lib/src/utils/accent_color_listener.dart

```diff
@@ -0,0 +1,160 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:appkit_ui_element_colors/appkit_ui_element_colors.dart';
+import 'package:flutter/foundation.dart';
+import 'package:macos_ui/src/enums/accent_color.dart';
+
+/// A class that listens to changes to the user’s selected system accent color.
+///
+/// Native macOS applications respond to such changes immediately.
+///
+/// Example using [StreamBuilder]:
+///
+/// ```dart
+/// StreamBuilder(
+///   stream: AccentColorListener.instance.onChangedStream,
+///     builder: (context, _) {
+///       final AccentColor? accentColor =
+///           AccentColorListener.instance.currentAccentColor;
+///
+///       return SomeWidget(
+///         accentColor: accentColor,
+///         child: …
+///       );
+///     },
+///  );
+/// ```
+class AccentColorListener {
+  /// A shared instance of [AccentColorListener].
+  static final instance = AccentColorListener();
+
+  /// A map which maps hue components of the [UiElementColor.controlAccentColor]
+  /// color captured with the [NSAppearanceName.aqua] appearance in the
+  /// [NSColorSpace.genericRGB] color space to the corresponding [AccentColor].
+  static final hueComponentToAccentColor = <double, AccentColor>{
+    0.6085324903200698: AccentColor.blue,
+    0.8285987697113538: AccentColor.purple,
+    0.9209523937489168: AccentColor.pink,
+    0.9861913496946438: AccentColor.red,
+    0.06543037411201169: AccentColor.orange,
+    0.11813830353929083: AccentColor.yellow,
+    0.29428158007138466: AccentColor.green,
+    0.0: AccentColor.graphite,
+  };
+
+  /// The currently active accent color.
```

Comment:
```suggestion
  /// The active accent color selection.
```
"Active" already implies currency